### DISCUSSION TITLE
isar is amazing (new benchmark)

### DISCUSSION THREAD

**Discussion by msxenon** [isar/isar#211] [Show and tell]
I have just checked using this benchmark
and I got objectbox (^1.4.0-dev.0)
was 1.5x faster than Isar DB Async (2.0.0); (with a little UI Lag)

BUT Isar Async is pretty cool while it's fast and not blocking the UI
<img width="465" alt="screenshot" src="https://user-images.githubusercontent.com/10958931/151562368-b69f2a65-9fda-46d8-b852-5116c4462a3e.png">

checkout the demo video and the screenshot below 
https://github.com/msxenon/flutter-db-benchmarks

**Comment by simc** [comment]
That's cool but the numbers seem a bit off. Isar sync should be much faster than async and very similar to objectbox. It's important that you don't wrap sync isar into a future. otherwise you won't get the correct results.

**Reply by msxenon** [reply]
alrighty, will double check that executors

**Reply by msxenon** [reply]
Well, I tried using “FutureOr” instead of “Future” and it was the same.
BTW in the benchmark they are all using the same parent class ExecutorBase and still objectbox is faster.

Finally: for using it in a production app, performance is more important than speed. So Isar Async here wins 🏆

**Reply by simc** [reply]
Interesting but I guess it makes sense that for batch operations there is not a big difference. On my device (ios sim), Isar is still quite similar to objectbox 🤷🏼‍♂️
![Simulator Screen Shot - iPhone 13 - 2022-02-01 at 10 55 31](https://user-images.githubusercontent.com/13610195/151947766-2eaea100-fff0-45c4-9807-e481a4772f95.png)

**Reply by msxenon** [reply]
Sure it differs from one device to another; I am testing on a low-end tablet to see the worst-case scenario

**Reply by simc** [reply]
that's a good hint. maybe there are some improvements that can be made for low-end devices 🤔
